python/tank/platform/software_launcher.py

```diff
@@ -304,50 +305,53 @@ def _format(self, template, tokens):
         return template
 
     def _glob_and_match(self, match_template, template_key_expressions):
-        """
+        r"""
         This is a helper method that can be invoked in an implementation of :meth:`scan_software`.
 
         The ``match_template`` argument provides a template to use both for globbing files and then pattern
         matching them using regular expressions provided by the ``tokens_expressions`` dictionary.
 
-        The method will first substitute every token surrounded by `{}` from the template with a ``*``
+        The method will first substitute every token surrounded by ``{}`` from the template with a ``*``
         for globbing files. It will then replace the tokens in the template with the regular expressions
         that were provided.
 
         In the following example::
 
             self._glob_and_match(
-                "//network/softwares/Nuke{full_version}/Nuke{major_minor_version}.exe",
+                "\\\\network\\softwares\\Nuke{full_version}\\Nuke{major_minor_version}.exe",
                 {
-                    "full_version": r"([\d.v]+)",
-                    "major_minor_version": r"([\d.]+)"
+                    "full_version": r"(?P<full_version>[\d.v]+)",
+                    "major_minor_version": r"(?P<major_minor_version>[\d.]+)"
                 }
             )
 
-        this would first look for every files matching C:/Program Files/Nuke*/Nuke*.exe and then
-        run the regular expression C:/Program Files/Nuke([\d.v]+)/([\d.]+)
-        and would return any files matching that pattern as well as any extracted tokens.
+        this would first look for every files matching the glob ``\\network\softwares\Nuke*\Nuke*.exe``
+        and then run the regular expression ``\\\\network\\softwares\\Nuke([\d.v]+)\\Nuke([\d.]+).exe``
+        on each matches. Each match will be returned with the accompanying regular expression groups and group
+        dictionary, if any were specified.
+
+        For example, if Nuke 10.0v1 was installed, the following would have been returned::
 
-        .. note::
-            The ``match_template`` argument must always use ``/`` for path separators. This allows to disambiguate
-            the meaning of `\\`` on Windows when matching executables with regular expressions. For example
-            if scannning for Maya installs inside the ``C:\Program Files`` folder for example, specify your template
-            as ``C:/Program Files/Maya{version}``.
+            [("\\\\network\\softwares\\Nuke10.0v1\\Nuke10.1.exe",
+              ("10.0v1", "10.0"),
+              {"full_version": "10.0v1", "major_minor_version"="10.0"})]
```

Comment:
Suggest adding a specific section that highlights the windows path issue and suggests always using raw strings as in the example above.